Frameworks/include/CACompositor.h

```diff
@@ -13,17 +13,102 @@
 // THE SOFTWARE.
 //
 //******************************************************************************
+#pragma once
 
-#if defined(__cplusplus) && defined(__OBJC__) && !defined(__CA_COMPOSITOR_H)
-#define __CA_COMPOSITOR_H
+#if defined(__OBJC__)
+
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wignored-attributes"
+
+#include <COMIncludes.h>
+#import <WRLHelpers.h>
+#import <ErrorHandling.h>
+#import <RawBuffer.h>
+#import <wrl/client.h>
+#import <wrl/implements.h>
+#import <wrl/async.h>
+#import <wrl/wrappers/corewrappers.h>
+#import <windows.foundation.h>
+#include <COMIncludes_end.h>
+
+#pragma clang diagnostic pop
+
+#else
 
 #include <memory>
-#include "winobjc\winobjc.h"
+#include <wrl/client.h>
+
+#endif
+
+// Proxy between CALayer and its Xaml representation
+struct ILayerProxy {
+public:
+    virtual ~ILayerProxy() {
+    }
+
+    // Returns the Xaml element backing this layer
+    virtual Microsoft::WRL::ComPtr<IInspectable> GetXamlElement() = 0;
+
+    // Returns the Xaml element used for hosting this layer's sublayer element
+    virtual Microsoft::WRL::ComPtr<IInspectable> GetSublayerXamlElement() = 0;
+
+    virtual void* GetPropertyValue(const char* propertyName) = 0;
+    virtual void SetShouldRasterize(bool shouldRasterize) = 0;
+
+    // TODO: Can we remove this altogether at some point?
+    virtual void SetTopMost() = 0;
+};
+
+// Proxy between CAAnimation and its Xaml representation
+struct ILayerAnimation {
+public:
+    virtual ~ILayerAnimation() {
+    }
+};
+
+// Proxy for CG-rendered content
+struct IDisplayTexture {
+public:
+    virtual ~IDisplayTexture() {
+    }
+
+    virtual Microsoft::WRL::ComPtr<IInspectable> GetContent() = 0;
+    virtual void* Lock(int* stride) = 0;
```

Comment:
That would require more churn than I think is worth it at this point, and at the end of the day, the value is an int (*stride = writableBitmap->PixelWidth * 4;)

---
In reply to: [86886920](https://github.com/Microsoft/WinObjC/pull/1318#discussion_r86886920) [](ancestors = 86886920)